app/routes.py

```diff
@@ -2,35 +2,110 @@
 from flask_login import login_user, logout_user, current_user, login_required
 from werkzeug.urls import url_parse
 from werkzeug import secure_filename
-from app import app, db
+from app import app, db, crypto, email
 from app.forms import LoginForm, RegistrationForm, VerifyForm, AdminAuthorizationForm
 from app.models import User, PPE, Hospital, Wants, Has, Exchanges, Exchange, EXCHANGE_COMPLETE, EXCHANGE_COMPLETE_TEXT, EXCHANGE_COMPLETE_ADMIN, EXCHANGE_COMPLETE_ADMIN_TEXT, EXCHANGE_COMPLETE_HOSPITAL_CANCELED, EXCHANGE_COMPLETE_HOSPITAL_CANCELED_TEXT, EXCHANGE_COMPLETE_ADMIN_CANCELED, EXCHANGE_COMPLETE_ADMIN_CANCELED_TEXT, EXCHANGE_UNVERIFIED, EXCHANGE_UNVERIFIED_TEXT, EXCHANGE_IN_PROGRESS, EXCHANGE_IN_PROGRESS_TEXT, EXCHANGE_NOT_ACCEPTED, EXCHANGE_ACCEPTED_NOT_SHIPPED, EXCHANGE_ACCEPTED_SHIPPED, EXCHANGE_ACCEPTED_RECEIVED, EXCHANGE_HOSPITAL_CANCELED, EXCHANGE_ADMIN_CANCELED, EXCHANGE_ADMIN_NOT_VERIFIED
 from app import crypto
 from app import email
 from datetime import datetime
-from sqlalchemy import desc
+from sqlalchemy import desc, and_, or_
+from sqlalchemy.orm import contains_eager
 
 import json
 import os
 import math
+import inspect
+
+
+import logging
+logging.basicConfig()
+# logging.getLogger('sqlalchemy.engine').setLevel(logging.INFO)
 
 @app.route('/')
 @app.route('/index')
 @login_required
 def index():
-    user = User.query.filter_by(username=current_user.username).first()
+    user = User.query.\
+        filter_by(username=current_user.username).\
+        join(Hospital).\
+        options(contains_eager(User.hospital)).\
+        first()
+
     if user.is_admin:
         return render_template('admin_index.html', title='Home')
     elif not user.is_verified:
         return render_template("404.html")
-    else:
-        user = User.query.filter_by(username=current_user.username).first()
-        user_hospital = Hospital.query.filter_by(id=user.hospital_id).first()
-        
-        hospital = {
-            "hospital_name": user_hospital.name
-        }
-        return render_template('index.html', title='Home', hospital=hospital)
+
+    ppe_types = PPE.query.\
+        outerjoin(Wants, and_(Wants.hospital_id == user.hospital.id, Wants.ppe_id == PPE.id)).\
```

Comment:
By making this an outer join, the filtering only applies to the `Wants`/`Has`. Therefore, it will always return _all_ `PPE`s, along with whatever `Wants`/`Has` that match the current hospital.